tower-balance/src/select/round_robin.rs

```diff
@@ -0,0 +1,64 @@
+use futures::{Async, Poll};
+use ordermap::IterMut;
+use std::hash::Hash;
+use std::marker::PhantomData;
+
+use {PollLoad,  Select};
+
+/// Selects nodes sequentially.
+///
+/// Note that ordering is not strictly enforced, especially when nodes are removed.
+pub struct RoundRobin<K, S>
+where
+    K: Hash + Eq,
+    S: PollLoad,
+{
+    /// References the index of the next node to be polled.
+    pos: usize,
+    _p: PhantomData<(K, S)>,
+}
+
+impl<K, S> Default for RoundRobin<K, S>
+where
+    K: Hash + Eq,
+    S: PollLoad,
+{
+    fn default() -> Self {
+        Self {
+            pos: 0,
+            _p: PhantomData,
+        }
+    }
+}
+
+impl<K, S> Select for RoundRobin<K, S>
+where
+    K: Hash + Eq,
+    S: PollLoad,
+{
+    type Key = K;
+    type Service = S;
+
+    fn poll_next_ready<'s>(&mut self, iter: IterMut<'s, K, S>) -> Poll<&'s Self::Key, S::Error> {
+        let mut nodes = iter.collect::<Vec<_>>();
```

Comment:
This is pretty unfortunate. I'd probably want to try to figure out a way to avoid the unnecessary overhead.

The only option that comes to mind right now is to pass in `endpoints: &[Entry<S>]` where `Entry` wraps the service and tracks if it is ready or not.

Reply:
We need to figure out what is responsible for polling readiness ;)  This is only possible if the balancer is responsible for managing the readiness state of all nodes -- at that point the balancer can just maintain a total ordering of all nodes weights; and we wouldn't even need to do P2C since we have an ordered heap.

This isn't the craziest approach, but i worry it won't stand up as we support more nodes.